lib/pure/times.nim

```diff
@@ -260,6 +260,7 @@ type
   Month* = enum ## Represents a month. Note that the enum starts at ``1``,
                 ## so ``ord(month)`` will give the month number in the
                 ## range ``1..12``.
+    # mInvalid = (0, "Invalid") # intentionally left out so `items` works
```

Comment:
If you leave January at 1, how come this PR fixes anything? DateTime should still be a type requiring explicit initialization then.

Reply:
already answered earlier in https://github.com/nim-lang/Nim/pull/14002#issuecomment-615195005

> Even with this PR default(DateTime) is invalid according to the type system since DateTime.month will be 0, so it's surprising that it even works.

it's currently legal in Nim and doesn't give any warnings; if it becomes illegal or deprecated, this can be changed

this PR does fix the monthday issue, and `var a: DateTime` does compile, see tests.
The month issue can be further addressed in future PRs, but this fixes a real usability issue.